tools/syncTranslations.py

```diff
@@ -43,25 +69,67 @@ def writeFile(path, keys, trans):
     lines = ''.join(lines) + '\n\n'
-    with open(path, 'w', encoding="utf8") as file:
-        file.write(lines)
-    return c
-        
-        
-showMissing = False     # output number of missing translations
-path = "../src/main/resources/resources/logisim/strings/"
-files = ("analyze","circuit", "data", "draw", "file", "fpga", "gui", "hdl", "proj", "soc", "std", "tools", "util")
-langs = ("de", "el", "es", "fr", "pt", "ru", "it", "nl", "ja", "pl")
-
-for file in files:
-    filePath = path + file +'/'+file
-    if showMissing:
-        print(file + ':')
-    keys = getKeys(filePath + ".properties")
-    for lang in langs:
-        trans = getTrans(filePath+'_'+lang+".properties")
-        missing = writeFile(filePath+'_'+lang+".properties", keys, trans)
-        if showMissing and missing > 0:
-            print("  " + lang + ": " + str(missing))
+    if (args.write):
+        with open(path, 'w', encoding = "utf8") as file:
+            file.write(lines)
+    return missingTranslations
+
+
+def main():
+    defaultRootDir = "../src/main/resources/resources/logisim/strings/"
+    files = sorted(["analyze", "circuit", "data", "draw", "file", "fpga", "gui", "hdl", "proj", "soc", "std", "tools", "util"])
+    maxFileNameLen = max([len(i) for i in files]) + 2
+    langs = sorted(["de", "el", "es", "fr", "pl", "pt", "ru", "it", "nl", "ja"])
+
+    parser = argparse.ArgumentParser()
+    group = parser.add_argument_group('Options')
+    group.add_argument('--write', action = 'store_true', dest = 'write',
+                       help = "Update translation files")
+    group.add_argument('-q', '--quiet', action = 'store_true', dest = 'quiet')
+
+    group = parser.add_argument_group('Paths')
+    group.add_argument('--root', action = 'store', dest = 'rootDir', nargs = 1, metavar = 'DIR',
+                       help = 'String resources root dir. Default: {}'.format(defaultRootDir))
+
+    args = parser.parse_args()
+
+    if args.rootDir is None:
+        args.rootDir = defaultRootDir
+
+    if not args.quiet:
+        h1 = ' ' * maxFileNameLen + '|'
+        h2 = '-' * maxFileNameLen + '+'
+        for lang in langs:
+            h1 += '{:^5} |'.format(lang.upper())
+        h2 += '------+' * len(langs)
+        print('{}\n{}'.format(h1, h2))
+
+    for file in files:
+        filePath = os.path.join(args.rootDir, file, file)
+        if not args.quiet:
+            fmt = '{:<' + str(maxFileNameLen) + '}|'
+            print(fmt.format(file), end = '')
+        keys = getKeys(filePath + ".properties")
+
+        failed = False
+        for lang in langs:
+            trans = getTrans(filePath + '_' + lang + ".properties")
+            missing = writeFile(args, filePath + '_' + lang + ".properties", keys, trans)
+            failed = failed or missing
+            if not args.quiet:
+                if missing == 0:
+                    missing = '-'
+                print(' {:>4} |'.format(missing), end = '')
```

Comment:
You should consider using the much cleaner [f-strings](https://docs.python.org/3/reference/lexical_analysis.html#f-strings).